PatternPal/PatternPal/Services/StepByStepService.cs

```diff
@@ -1,41 +1,211 @@
-﻿using PatternPal.Core.StepByStep;
+﻿#region
+
+using System.Reflection;
+
+using PatternPal.Core.Recognizers;
+using PatternPal.Core.StepByStep;
 using PatternPal.SyntaxTree;
 using PatternPal.SyntaxTree.Abstractions.Entities;
 
+using InstructionSet = PatternPal.Core.StepByStep.InstructionSet;
+
+#endregion
+
 namespace PatternPal.Services;
 
 public class StepByStepService : Protos.StepByStepService.StepByStepServiceBase
 {
+    /// <inheritdoc />
     public override Task< GetInstructionSetsResponse > GetInstructionSets(
         GetInstructionSetsRequest request,
         ServerCallContext context)
     {
-        throw new NotImplementedException();
+        GetInstructionSetsResponse response = new();
+
+        // Obtain all the enum values in the proto.
+        bool initialValue = true;
```

Comment:
This can be improved by using the `RecognizerRunner.SupportedRecognizers.Keys` collection, which was not available in the past. I'd also move this method to a helper class, so that it can be reused by the `RecognizerService`.

Reply:
I have made use of the supported recognizers in the runner and filter accordingly using a try catch.